packages/amplify_flutter/lib/amplify.dart

```diff
@@ -27,51 +30,50 @@ import 'categories/amplify_categories.dart';
 
 part 'method_channel_amplify.dart';
 
-class Amplify extends PlatformInterface {
-  static const AuthCategory Auth = const AuthCategory();
-  static const AnalyticsCategory Analytics = const AnalyticsCategory();
-  static const StorageCategory Storage = const StorageCategory();
-  static const DataStoreCategory DataStore = const DataStoreCategory();
+/// Top level singleton Amplify object.
+final AmplifyClass Amplify = new AmplifyClass._private();
+
+class AmplifyClass extends PlatformInterface {
+  AmplifyClass._private();
+
```

Comment:
This `_private()` method is a factory constructor to ensure a singleton?

Reply:
yes, a named constructor starting with a `_` is private to the library (like any other identifier in dart) and so cannot be initialized by the apps. This ensure that only one instance exists for this class which is accessed by the static `Amplify`.